src/Core/Module/SourceHandler/ZipSourceHandler.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Module\SourceHandler;
+
+use PrestaShop\PrestaShop\Core\Module\Exception\ModuleErrorException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use ZipArchive;
+
+class ZipSourceHandler implements SourceHandlerInterface
+{
+    private const AUTHORIZED_MIME = [
+        'application/zip',
+        'application/x-gzip',
+        'application/gzip',
+        'application/x-gtar',
+        'application/x-tgz',
+    ];
+
+    private const MODULE_REGEX = '/^(.*)\/\1\.php$/i'; // module_name/module_name.php
+
+    /** @var string */
+    private $modulePath;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    public function __construct(string $modulePath, TranslatorInterface $translator)
+    {
+        if (substr($modulePath, -1) !== '/') {
+            $modulePath .= '/';
+        }
```

Comment:
Alternative to avoid testing the path provided as argument, and to prevent duplicated slashes:
```suggestion
        $modulePath = rtrim($modulePath, '/') . '/';
```